components/script/security_manager.rs

```diff
@@ -99,7 +99,8 @@ impl CSPViolationReportTask {
         for token in &report_uri_directive.value {
             // Step 3.4.2.1. Let endpoint be the result of executing the URL parser with token as the input,
             // and violation’s url as the base URL.
-            let Ok(endpoint) = ServoUrl::parse_with_base(Some(&global.get_url()), token) else {
+            let Ok(endpoint) = ServoUrl::parse_with_base(Some(&global.api_base_url()), token)
```

Comment:
Using the API base url here correctly. Again, the spec is ambiguous here where it says that the violation URL is the global URL, despite the spec also saying it should use the API base URL instead.